cognite/neat/graph/transformations/query_generator/sparql.py

```diff
@@ -283,3 +290,246 @@ def build_sparql_query(
+
+    # add first triple for graph pattern stating type of object
+    patterns += [
+        Triple(
+            subject="?subject",
+            predicate="a",
+            object=_most_occurring_element([property_.rule.class_.id for property_ in properties]),
+            optional=False,
+        )
+    ]
+
+    for property_ in properties:
+        graph_template_triple = Triple(
+            subject="?subject",
+            predicate=f"{transformation_rules.metadata.prefix}:{property_.property_id}",
+            object=f'?{re.sub(r"[^_a-zA-Z0-9/_]", "_", str(property_.property_id).lower())}',
+            optional=False,
+        )
+
+        if isinstance(property_.rule, AllReferences):
+            graph_pattern_triple = Triple(
+                subject="BIND(?subject",
+                predicate="AS",
+                object=f"{graph_template_triple.object})",
+                optional=True if properties_optional else not property_.mandatory,
+            )
+
+        elif isinstance(property_.rule, SingleProperty):
+            graph_pattern_triple = Triple(
+                subject=graph_template_triple.subject,
+                predicate=property_.rule.property.id,
+                object=graph_template_triple.object,
+                optional=True if properties_optional else not property_.mandatory,
+            )
+
+        elif isinstance(property_.rule, Hop):
+            graph_pattern_triple = Triple(
+                subject="?subject",
+                predicate=_hop2property_path(graph, property_.rule, transformation_rules.prefixes),
+                object=graph_template_triple.object,
+                optional=True if properties_optional else not property_.mandatory,
+            )
+        else:
+            continue
+
+        patterns += [graph_pattern_triple]
+        templates += [graph_template_triple]
+
+    return templates, patterns
+
+
+def _count_element_occurrence(list_of_elements: list):
+    occurrence: dict[str, int] = {}
+    for i in list_of_elements:
+        if i in occurrence:
+            occurrence[i] += 1
+        else:
+            occurrence[i] = 1
+    return occurrence
+
+
+def _most_occurring_element(list_of_elements: list):
+    occurrence = _count_element_occurrence(list_of_elements)
+    return max(occurrence, key=occurrence.get)
+
+
+def triples2dictionary(triples: list[tuple[URIRef, URIRef, str | URIRef]]) -> dict[str, list[str]] | dict[str, str]:
+    """Converts list of triples to dictionary"""
+    dictionary = {}
+    for triple in triples:
+        id_ = remove_namespace(triple[0])
+        property_ = remove_namespace(triple[1])
+        value = remove_namespace(triple[2])
+
+        if id_ not in dictionary:
+            dictionary["external_id"] = [id_]
+
+        if property_ in dictionary:
+            dictionary[property_].append(value)
+        else:
+            dictionary[property_] = [value]
+    return dictionary
```

Comment:
A case for defaultdict
```suggestion
    dictionary = defaultdict(list)
    for triple in triples:
        id_ = remove_namespace(triple[0])
        property_ = remove_namespace(triple[1])
        value = remove_namespace(triple[2])

        if id_ not in dictionary:
            dictionary["external_id"] = [id_]
        
        dictionary[property_].append(value)
    return dictionary
```